controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go

```diff
@@ -259,6 +267,99 @@ func (r *AKODeploymentConfigReconciler) reconcileCloudUsableNetwork(
 	return res, nil
 }
 
+func (r *AKODeploymentConfigReconciler) reconcileAviInfraSetting(
+	ctx context.Context,
+	log logr.Logger,
+	adc *akoov1alpha1.AKODeploymentConfig,
+) (ctrl.Result, error) {
+	res := ctrl.Result{}
+
+	log.Info("Start reconciling AVIInfraSetting")
+
+	if adc.Spec.ControlPlaneNetwork.Name == "" {
+		log.Info("ControlPlaneNetwork is empty in akoDeploymentConfig, skip creating AVIInfraSetting")
+		return res, nil
+	}
+
+	newAviInfraSetting := r.createAviInfraSetting(adc)
+
+	if err := r.Get(ctx, client.ObjectKey{
+		Name: haprovider.GetAviInfraSettingName(adc),
+	}, newAviInfraSetting); err != nil {
```

Comment:
You updated newAviInfraSetting with the old one here, then update old with new aviinfrasetting on line 296, which makes no sense. It should be something like this:
```go
	newAviInfraSetting = r.createAviInfraSetting(adc)
	aviInfraSetting := &AVIInfraSetting{}
	if err := r.Get(ctx, client.ObjectKey{
		Name: haprovider.GetAviInfraSettingName(adc),
	}, aviInfraSetting); err != nil {
		if apierrors.IsNotFound(err) {
			log.Info("AVIInfraSetting doesn't exist, start creating it")
			return res, r.Create(ctx, newAviInfraSetting)
		} else {
			log.Error(err, "Failed to get AVIInfraSetting, requeue")
			return res, err
		}
	} else {
		log.Info("Update AVIInfraSetting %s", aviInfraSetting.name)
		aviInfraSetting = newAviInfraSetting.DeepCopy()	
		return res, r.Update(ctx, aviInfraSetting)
	}
```

Reply:
Good catch! Thanks Xudong!